packages/back-end/src/integrations/BigQuery.ts

```diff
@@ -109,4 +111,41 @@ export default class BigQuery extends SqlIntegration {
       database
     );
   }
+  async getInformationSchema(): Promise<InformationSchema[]> {
+    const datasets = await this.runQuery(
+      `SELECT * FROM ${`\`${this.params.projectId}.INFORMATION_SCHEMA.SCHEMATA\``}`
+    );
+
+    const query: string[] = [];
+
+    for (const dataset of datasets) {
+      query.push(`
```

Comment:
Should we instead run these queries side-by-side and join the results together in the method here instead of running one job? Seems like it could be more performant, but it's probably not a big deal either way.

Then, eventually when you also do this for, say, DataBricks catalogs, it would be easy to abstract this method and have the base query live in only one place (but maybe the abstraction isn't worth it at this point).

Reply:
@lukesonnet Re: running the queries side by side & joining the results, are you suggesting something like...

```ts
const results = [];

    for (const dataset of datasets) {
      const query = `SELECT
        table_name as table_name,
        table_catalog as table_catalog,
        table_schema as table_schema,
        count(column_name) as column_count
      FROM
        ${this.getInformationSchemaTable(dataset.schema_name)}
        WHERE ${this.getInformationSchemaWhereClause()}
      GROUP BY table_name, table_schema, table_catalog
      ORDER BY table_name;`;

      const datasetResults = await this.runQuery(
        format(query, this.getFormatDialect())
      );

      if (datasetResults.length > 0) {
        results.push(...datasetResults);
      }
    }

    if (!results.length) {
      throw new Error(`No tables found.`);
    }

    return formatInformationSchema(
      results as RawInformationSchema[],
      this.type
    );
 ```

I think for the abstraction, I'd rather wait until we're doing the DataBricks stuff, just because there are some unknowns that I'll likely need to account for that I won't really know until I'm working through the DataBricks updates.

Reply:
FWIW - The code above works and I have it staged, just want to be sure that's what you had in mind.

Reply:
Yeah something like that, it also might be less brittle in case one of the queries fails? Although IDK if we handle that gracefully (and it's probably not worth building out now). But IDK, this one is up to you and your experience with these queries, just wanted to throw it out there.

Agreed re: abstraction, fine with me.